### PR TITLE
Add esm module entry point to package

### PR DESCRIPTION
**Pull request by shinobi5**
Issue #15 

Change `main` to `module` in `package.json` to provide an `esm` module entry point into the package.

There's only one format (esm) for the output generated by rollup so I didn't keep both `main` and `module` properties.

### CODE REVIEW

**Comment by shinobi5** [comment]
Thinking about this some more and we could generate a cjs output as well and set that to `main`? Depends how you want to approach this. Do you see anyone wanting to use this with node and using `require` to import it?

**Comment by mkay581** [comment]
This looks great! Thank you so much! Yeah, I definitely agree we should allow importing it via `require()`. I know its rather old-school, but I think there may be a lot of consumers who may still need to consume the library in such a way. Do you mind updating rollup to produce a commonjs output as well and adding it to `main`?

**Comment by shinobi5** [comment]
Agree. I'll get that sorted.

File: package.json [hunk 1]
```diff
@@ -41,7 +42,7 @@
     "mocha": "^5.2.0",
     "rollup": "^0.67.0",
     "rollup-plugin-node-resolve": "^3.4.0",
-    "rollup-plugin-typescript2": "^0.18.1",
+    "rollup-plugin-typescript2": "^0.21.1",
```
Comment (mkay581) [inline package.json]: love this thanks!

**Comment by mkay581** [comment]
This has been released in latest version 0.8.0!

**Comment by shinobi5** [comment]
Nice one, thanks @mkay581 !

Thanks for this component, it's great.

**Pull request merged**